.jenkins/library/vars/helpers.groovy

```diff
@@ -0,0 +1,317 @@
+
+/****************************************
+* Shared Library for Helpers and Commands
+****************************************/
+
+def CmakeArgs(String build_type = "RelWithDebInfo", String code_coverage = "OFF", String debug_malloc = "ON", String lvi_args="", String cmake_args = "") {
+    def args = "-G Ninja -DCMAKE_INSTALL_PREFIX:PATH='/opt/openenclave' -DCPACK_GENERATOR=DEB -DCODE_COVERAGE=${code_coverage} -DUSE_DEBUG_MALLOC=${debug_malloc} -DCMAKE_BUILD_TYPE=${build_type} ${lvi_args} ${cmake_args} -Wdev"
+    return args
+}
+
+def WaitForAptLock() {
+    def aptWait = """
+        i=0
+        echo "Checking for locks..."
+        # Check for locks
+        while fuser /var/lib/dpkg/lock > /dev/null 2>&1 ||
+              fuser /var/lib/dpkg/lock-frontend > /dev/null 2>&1 ||
+              fuser /var/lib/apt/lists/lock > /dev/null 2>&1; do
+            # Wait up to 600 seconds to lock to be released
+            if (( i > 600 )); then
+                echo "Timeout waiting for lock."
+                exit 1
+            fi
+            echo "Waiting for apt/dpkg locks..."
+            i=\${i++}
+            sleep 1
+        done
+    """
+    return aptWait
+}
+
+def TestCommand() {
+    def testCommand = """
+        echo "Running Test Command"
+        ctest --output-on-failure --timeout ${globalvars.CTEST_TIMEOUT_SECONDS}
+    """
+    return testCommand
+}
+
+def InstallBuildCommand() {
+    def installCommand = """
+        echo "Running Install Build Command"
+        cpack -G DEB
+        ${WaitForAptLock()}
+        sudo ninja -v install
+    """
+    return installCommand
+}
+
+def InstallReleaseCommand() {
+    def installCommand = """
+        echo "Running Install Release Command"
+        echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+        wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+
+        echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main" | sudo tee /etc/apt/sources.list.d/llvm-toolchain-bionic-7.list
+        wget -qO - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+
+        echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main" | sudo tee /etc/apt/sources.list.d/msprod.list
+        wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+
+        ${WaitForAptLock()}
+        sudo apt update
+
+        ${WaitForAptLock()}
+        sudo apt install -y open-enclave
+
+        echo "Open Enclave SDK version installed"
+        apt list --installed | grep open-enclave
+    """
+    return installCommand
+}
+
+/**
+ * Tests Open Enclave samples on Unix and Windows systems
+ *
+ * @param lvi_mitigation  Determines whether tests should be ran with LVI mitigation
+ */
+def TestSamplesCommand(boolean lvi_mitigation = false, String oe_package = "open-enclave") {
```

Comment:
I like that this function automatically handles different OS versions, it might be cleaner to put the Linux tests in a separate function (maybe `TestSamples_Linux`), so the function is simplified to:

```
def TestSamplesCommand(boolean lvi_mitigation = false, String oe_package = "open-enclave") {
  if (isUnix()) {
    TestSamples_Linux()
  } else {
    TestSamples_Windows()
  }
}
```

Likewise for `ninjaBuildCommand`. Definitely open for discussion as this is purely preference.

Reply:
Added `testSamplesWindows`, `testSamplesLinux`, `releaseInstallWindows`, and `releaseInstallLinux`. Renamed `releaseInstallCommand` to `releaseInstall`